pennylane/decomposition/decomposition_graph.py

```diff
@@ -162,10 +164,24 @@ def _get_decompositions(self, op_node: CompressedResourceOp) -> list[Decompositi
 
         decomps = self._alt_decomps.get(op_name, []) + list_decomps(op_name)
 
-        if issubclass(op_node.op_type, qml.ops.Adjoint):
+        if (
+            issubclass(op_node.op_type, qml.ops.Adjoint)
+            and self_adjoint not in decomps
+            and adjoint_rotation not in decomps
+        ):
+            # Only when the base op is not self-adjoint or a rotation with a single rotation
+            # angle that could be trivially negated do we need to apply adjoint to each of
+            # the base op's decomposition rules.
```

Comment:
```suggestion
            # We only need to apply adjoint to each of the base op's decomposition rules
            # if the base op is not self-adjoint or a rotation gate with a single angle
            # (for which we can simply flip the sign of the rotation angle)
```

Based on the explanation that you provided, I tried to re-formulate the sentence in a way that was more intuitive for me to understand. If there is something incorrect or you don't like this version, please let me know!

Reply:
I elaborated the reasoning even further. Can you take a look at how it's phrased now?

Reply:
Amazing!